fmlloader/src/main/java/net/minecraftforge/fml/loading/targets/UnionHelper.java

```diff
@@ -0,0 +1,51 @@
+package net.minecraftforge.fml.loading.targets;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.spi.FileSystemProvider;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.function.BiPredicate;
+
+import org.jetbrains.annotations.ApiStatus;
+
+@ApiStatus.Internal
+final class UnionHelper {
+    private static final FileSystemProvider UFSP = getUnionFileSystemProvider();
+    private static FileSystemProvider getUnionFileSystemProvider() {
+        for (var provider : FileSystemProvider.installedProviders()) {
+            if ("union".equals(provider.getScheme()))
+                return provider;
+        }
+        throw new IllegalStateException("Missing UnionFileSystemProvider");
+    };
+
+    static FileSystem newFileSystem(BiPredicate<String, String> filter, Path[] paths) {
+        try {
+            if (paths == null || paths.length == 0)
+                throw new IllegalArgumentException("Must contain atleast one path");
+
+            if (filter == null && paths.length == 1)
+                return FileSystems.newFileSystem(paths[0]);
+
+            var map = new HashMap<String, Object>();
+            if (filter != null)
+                map.put("filter", filter);
+
+            var lst = new ArrayList<>(Arrays.asList(paths));
```

Comment:
Possibly unnecessary double-wrapping of paths - `Arrays.asList()` basically does a `new ArrayList<>()` internally.
```suggestion
            var lst = Arrays.asList(paths);
```

Reply:
Arrays.asList is immutable

Reply:
My mistake, I got confused because it creates an object of the same class name that's actually different to `java.util.ArrayList`.

Strangely, it does actually allow mutating its elements, but not changing its size:
<img width="368" alt="image" src="https://github.com/MinecraftForge/MinecraftForge/assets/3158390/8970fff0-a610-4678-bcc6-58ca18c64c40">

Reply:
Either way, I need to remove the first entry so giving myself a known good list implementation works.